Fix for issue 2

OscarTopliss opened the pull request:
Made updating the theme into a function and then added that function to the main loop. The script should check the theme each loop and update accordingly. I don't think this is the most efficient way to do this, but it seems to work. #2

OscarTopliss committed:
Fix for issue 2

Made updating the theme into a function and then added that function to the main loop. The script should check the theme each loop and update accordingly. I don't think this is the most efficient way to do this, but it seems to work.

NotALakeLurk commented:
Hey, thanks for helping. I had similar thoughts about the efficiency of checking the theme every millisecond. I did some testing and it seems to me that checking the theme every 0.1-2 seconds would be satisfactory for me, and much more performance friendly (even more efficient than expected for longer simulated times; though this could easily be a fault of my tests). 

I believe that the above should be easily achievable by adding some form of the below code to the canvas updater function.

```
function drawDateToCanvas(...) {
  ...
  let i = 0;
  ...
  while(true) {
    if (i >= checkThemeTimeoutMs) { updateTheme(); }
    ...
    await new Promise(...);
    i++;
  }
}
```

Thank you for your consideration.

OscarTopliss commented:
no problem! that sounds like a good idea to me, I'll add it in.

OscarTopliss committed:
Reduced themeUpdate() frequency

Reduced the frequency that the script checks for a theme update, as mentioned at #3

hemthead reviewed (approved):
Nice, looks good to me!
